File: OpenStash/Assets/Plugins/OpenStash/OSStats.js
#pragma strict

public class OSStats extends MonoBehaviour {
	public var name : String;
	public var hp : float = 100;
	public var maxHp : float = 100;
	public var mp : float = 100;
	public var maxMp : float = 100;
	public var attributes : OSAttribute[] = new OSAttribute [0];

	public function GetValue ( attrName : String ) : int {
		for ( var a : OSAttribute in attributes ) {
			if ( a.name == attrName ) {
				return a.value;
			}
		}

		return -1;
	}
}

File: OpenStash/Assets/Plugins/OpenStash/OSStats.cs
using UnityEngine;
using System.Collections;

public class OSStats : MonoBehaviour {
	public string name;
	public float hp = 100;
	public float maxHp = 100;
	public float mp = 100;
	public float maxMp = 100;
	public OSAttribute[] attributes = new OSAttribute [0];

	public float GetValue ( string attrName ) {
		foreach ( OSAttribute a in attributes ) {
			if ( a.name == attrName ) {
				return a.value;
			}
		}

		return -1;
	}
}
